plan/policy_test.go

```diff
@@ -64,6 +64,52 @@ func TestApply(t *testing.T) {
 	}
 }
 
+func TestApplyOwnershipPolicy(t *testing.T) {
+	// empty list of records
+	empty := []*endpoint.Endpoint{}
+	// a simple entry
+	fooV1 := []*endpoint.Endpoint{{DNSName: "foo", Target: "v1", Labels: map[string]string{"owner": "foo"}}}
+	// the same entry but with different target
+	fooV2 := []*endpoint.Endpoint{{DNSName: "foo", Target: "v2", Labels: map[string]string{"owner": "foo"}}}
+	// another two simple entries
+	bar := []*endpoint.Endpoint{{DNSName: "bar", Target: "v1", Labels: map[string]string{"owner": "foo"}}}
+	baz := []*endpoint.Endpoint{{DNSName: "baz", Target: "v1", Labels: map[string]string{"owner": "foo"}}}
+
+	for _, tc := range []struct {
+		policy   Policy
+		changes  *Changes
+		expected *Changes
+	}{
+		{
+			// SyncPolicy doesn't modify the set of changes.
+			&OwnershipPolicy{Owner: "foo"},
+			&Changes{Create: baz, UpdateOld: fooV1, UpdateNew: fooV2, Delete: bar},
+			&Changes{Create: baz, UpdateOld: fooV1, UpdateNew: fooV2, Delete: bar},
+		},
+		{
+			// SyncPolicy doesn't modify the set of changes.
+			&OwnershipPolicy{Owner: "bar"},
+			&Changes{Create: baz, UpdateOld: fooV1, UpdateNew: fooV2, Delete: bar},
+			&Changes{Create: empty, UpdateOld: empty, UpdateNew: empty, Delete: empty},
```

Comment:
basically drops every endpoint change if the endpoint is not owned by us. Note, create will never have ownership  information I think. @ideahitme right?